### DISCUSSION TITLE
WP Multisite Question

### DISCUSSION THREAD

**Discussion by Stonebkfly** [oidc-wp/openid-connect-generic#352] [Q&A] [answered]
When configuring openid-connect-generic for multisite what is a good way to do this? Each site has its own OpenID Connect Client settings with its own "Redirect URI". Does this mean that the SSO needs a client per site under the multisite or can all share the same SSO Client. I tried making a client per site but this caused a problem at logout when a user logs in at one site and then logs out at a different site as the PostLoggoutRedirectUri was not recognized by the SSO. I think it may be cleaner to have a single SSO Client for all sites but not sure if this recommended.

**Comment by timnolte** [comment]
@Stonebkfly this is a good question. I don't believe I have tested out a setup using a WordPress Mutlisite. I'll need to add that to the things to test out and try to update the Wiki to provide some guidance on this aspect.

**Reply by Stonebkfly** [reply]
Thank you Tim!

**Reply by ghost** [reply]
I'd like to second this option.  I know I'm late to this thread but am in a similar situation to @Stonebkfly where we have about a hundred sites that we use a single ID server to authenticate and I'd rather NOT give site admins control over the OIDC options.  Wondering if it's something we can set in say a sunrise.php file?

**Reply by timnolte** [reply]
FYI, if you want to restrict control of the settings this can be configured via constants in the `wp-config.php`. I really need to get these put in the Wiki but they are all documented here: https://github.com/oidc-wp/openid-connect-generic/blob/dev/HOWTO.md#configuration-environment-variablesconstants

**Reply by ghost** [reply]
Hey Tim
That's amazing.  Thanks for the link.  Do you see a difference between adding the constants to wp-config.php vs sunrise.php?
Best

**Reply by timnolte** [reply]
@chadsterBAM either will work just as well. They will be available sooner to WordPress if they are in the `sunrise.php` though our plugin will work the same even if they are just put in the `wp-config.php`.

**Comment by timnolte** [comment, marked as answer]
As an update I did get the configuration details added to the Wiki so hopefully that will make them more visible: https://github.com/oidc-wp/openid-connect-generic/wiki/Configuration-Environment-Variables-Constants
